app/controllers/postgresql_lo_streamer/lo_controller.rb

```diff
@@ -2,78 +2,27 @@
 
 module PostgresqlLoStreamer
   class LoController < ActionController::Base
+
     def stream
-      send_file_headers!(headers_for_extension_or_default)
 
-      object_identifier = params[:id].to_i
-      if !object_exists?(object_identifier)
-        self.status = 404
-        render :nothing => true and return
+      streamer = PostgresqlLoStreamer::Streamer.new(
+        connection,
+        params[:id].to_i,
+        params[:format]
+      )
+      send_file_headers!(streamer.default_headers)
```

Comment:
If you want to make the `default_headers` logic reusable maybe it could be a method at the `PostgresqlLoStreamer` module level. Something like:

```ruby
PostgresqlLoStreamer.default_headers_for(params[:format])
```